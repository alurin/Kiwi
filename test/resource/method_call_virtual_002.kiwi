#!/usr/bin/env kiwi
/* Test script for call virtual method from other methods + up cast  */

class A {
    void test(int $a) {
        print "Integer: ";
        print $a;
        print "\n";
    }
}

class B {
    inherit A;

    void test(int $a) {
        print "Override: ";
        print $a;
        print "\n";
    }
}

void test(A $var, int $a) {
    $var.test($a);
}

/// Main function
void main() {
    print "Main worked\n";
    test(new A(), 10);
    test(new B(), 11);
}
