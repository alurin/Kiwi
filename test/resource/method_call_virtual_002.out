Main worked
Integer: 10
Override: 11
